osutil/epoll/epoll_test.go

```diff
@@ -0,0 +1,411 @@
+package epoll_test
+
+import (
+	"testing"
+	"time"
+
+	"golang.org/x/sys/unix"
+
+	"github.com/snapcore/snapd/osutil/epoll"
+
+	. "gopkg.in/check.v1"
+)
+
+func Test(t *testing.T) { TestingT(t) }
+
+type epollSuite struct{}
+
+var _ = Suite(&epollSuite{})
+
+func (*epollSuite) TestString(c *C) {
+	c.Check(epoll.Readable.String(), Equals, "Readable")
+	c.Check(epoll.Writable.String(), Equals, "Writable")
+	c.Check(epoll.Readiness(epoll.Readable|epoll.Writable).String(), Equals, "Readable|Writable")
+}
+
+func (*epollSuite) TestOpenClose(c *C) {
+	e, err := epoll.Open()
+	c.Assert(err, IsNil)
+	c.Assert(e.Fd() == -1, Equals, false)
+	c.Assert(e.Fd() == 0, Equals, false)
+	c.Assert(e.Fd() == 1, Equals, false)
+	c.Assert(e.Fd() == 2, Equals, false)
+	c.Assert(e.RegisteredFdCount(), Equals, 0)
+
+	err = e.Close()
+	c.Assert(err, IsNil)
+	c.Assert(e.Fd(), Equals, -1)
+}
+
+func concurrentlyRegister(e *epoll.Epoll, fd int, errCh chan error) {
+	err := e.Register(fd, epoll.Readable)
+	errCh <- err
+}
+
+func concurrentlyDeregister(e *epoll.Epoll, fd int, errCh chan error) {
+	err := e.Deregister(fd)
+	errCh <- err
+}
+
+func waitMillisecondsThenWriteToFile(msec int, fd int, msg []byte) error {
```

Comment:
Super nitpick: maybe instead of ToFile -> ToFd ?

Reply:
Definitely, good catch!